lib/item.rb

```diff
@@ -0,0 +1,45 @@
+require 'date'
+require_relative 'genre'
+
+class Item
+  attr_accessor :id, :publish_date, :label, :author
+  attr_reader :genre
+
+  def initialize(date)
+    @id = Random.rand(1..1000)
+    @publish_date = Date.parse(date)
+    @archived = false
+    @genre = nil
+    @author = nil
+  end
+
+  def add_genre(genre)
+    if genre.instance_of?(Genre)
+      @genre = genre
+      genre.add_items(self) unless genre.items.include?(self)
+    else
+      @genre = nil
+    end
+  end
+
+  def add_author(author)
+    @author = author
+    author.items.push(self) unless author.items.include?(self)
+  end
+
+  def add_label(label)
+    @label = label
+    label.add_item(self) unless label.items.include?(self)
+  end
+
+  def move_to_archive
+    @archived = can_be_archived?
+  end
+
+  private
+
+  def can_be_archived?
+    current_date = Date.today
+    current_date - @publish_date >= 3652
```

Comment:
Hi @FrederickMih, thank you for pointing this out. Please kindly explain why converting 10 years into days is not an acceptable approach.
 
My logic behind doing so was that because each `publish_date` is accurate to the day, and we want to find out if that date is 10 years before the day when the comparison is being made, then it would make sense to measure the time in days. Additionally, the comparison I was making was to match the result from the operation the `Date` object is performing on the two dates. Ruby itself seems to be interpreting the difference between two dates as the number of days, so I provided it with the number of days in 10 years (including 2 leap years) but rounded down from 3652.5 for convenience's sake, and it seemed to work for us while staying within the parameter requirements.

If I made an error at any point in my reasoning, I kindly request that you point it out so that I will be mindful of it next time I face a similar challenge.

Reply:
@badger-99, kindly check my comment once more. I didn't mention that converting years into days is not an acceptable approach. My problem was that you are comparing two entities which are not the same. If publish date is greater than 10 years, I think it implies we are we can confidently work with years. Date.today gives a string not a unit as days. I don't know if I'm making sense here

Reply:
current_date and @publish_date is a sting of year, month and day. It is possible to compare it with days?. You should have convert 10 years into the same format(year, month, day)

Reply:
The Date class does return a string representation of the date object, but according to the [documentation](https://docs.ruby-lang.org/en/master/Date.html#method-i-2D) it can also find the difference between two date objects and return an integer value. It's how I am able to perform the comparison. If the two date objects have days and months, then the return value is the number of days between the two dates.